### PR TITLE
everywhere: replace double words

### PR DESCRIPTION
**Pull request by JordanYates**
Treewide search and replace on a range of double word combinations:
    * `the the`
    * `to to`
    * `if if`
    * `that that`
    * `on on`
    * `is is`
    * `from from`

Add double prepositions to the default spelling check list.

### CODE REVIEW

**Review by nashif — DISMISSED** [review]

**Review by aescolar — DISMISSED** [review]

**Comment by LingaoM** [comment]
Learn by analogy: there are also `to to`, `that that`, `for for`, etc...

**Review by LingaoM — CHANGES_REQUESTED** [review]
`to to`, `if if`, `that that`, `on on`, `is is`
So I recommend filtering all words.

**Comment by kartben** [comment]
I'm with @LingaoM , let's do even more if you don't mind, @JordanYates :)

You may also want to search more smartly, something like `the\n\s+(?:\* )?the ` will give you things like:

* https://github.com/zephyrproject-rtos/zephyr/blob/7f0f0a4f97d6adb6c134db0788dd200fbfba8515/drivers/sensor/st/lis2dh/Kconfig#L158-L160
* https://github.com/zephyrproject-rtos/zephyr/blob/7f0f0a4f97d6adb6c134db0788dd200fbfba8515/include/zephyr/crypto/crypto.h#L476-L478

**Comment by aescolar** [comment]
> to to, if if, that that, on on, is is

Please note that a PR should not be blocked to demand "more" fixes (https://docs.zephyrproject.org/latest/contribute/contributor_expectations.html#reviewer-expectations9.
If @JordanYates is kind enough to take it further that would be certainly very nice of him.
But this PR is improving something, and we cannot demand from contributors they spend their time fixing even more things just because we realize about them.

**Review by LingaoM — DISMISSED** [review]

**Review by carlescufi — DISMISSED** [review]

**Comment by JordanYates** [comment]
Updates to fix more double prepositions, did a manual pass across the changed lines to fix some capitalization and the occasional nonsensical sentance

**Review by kartben — DISMISSED** [review]
cool cool ❤️

File: dts/common/nordic/nrf91xx_partition.dtsi
Comment (aescolar) [inline dts/common/nordic/nrf91xx_partition.dtsi:1]: Did this just sneak in?

Comment (JordanYates) [reply dts/common/nordic/nrf91xx_partition.dtsi:1]: Haha whoops, a sneak peak of what I was doing before I found these

Comment (JordanYates) [reply dts/common/nordic/nrf91xx_partition.dtsi:1]: removed

**Review by aescolar — DISMISSED** [review]

**Review by kartben — DISMISSED** [review]

**Comment by nashif** [comment]
hopefully this did not remove any valid `wink wink` occurances :-D

**Review by nashif — DISMISSED** [review]

**Comment by Thalley** [comment]
Would it make sense to add this sort of check to CI?

File: tests/bluetooth/audio/cap_commander/src/test_micp.c [hunk 1]
```diff
@@ -134,7 +134,7 @@ ZTEST_F(cap_commander_test_micp, test_commander_change_microphone_gain_setting_d
 	zexpect_call_count("bt_cap_commander_cb.microphone_gain_setting_changed", 1,
 			   mock_cap_commander_microphone_gain_changed_cb_fake.call_count);
 
-	/* That that it still works as expected if we set the same value twice */
+	/* That it still works as expected if we set the same value twice */
```
Comment (Thalley) [inline tests/bluetooth/audio/cap_commander/src/test_micp.c:137]:
```suggestion
	/* Test that it still works as expected if we set the same value twice */
```

File: tests/bluetooth/audio/cap_commander/src/test_micp.c [hunk 2]
```diff
@@ -344,7 +344,7 @@ ZTEST_F(cap_commander_test_micp, test_commander_change_microphone_mute_state_dou
 	zexpect_call_count("bt_cap_commander_cb.microphone_mute_changed", 1,
 			   mock_cap_commander_microphone_mute_changed_cb_fake.call_count);
 
-	/* That that it still works as expected if we set the same value twice */
+	/* That it still works as expected if we set the same value twice */
```
Comment (Thalley) [inline tests/bluetooth/audio/cap_commander/src/test_micp.c:347]:
```suggestion
	/* Test that it still works as expected if we set the same value twice */
```

**Review by Thalley — CHANGES_REQUESTED** [review]

**Review by jhedberg — APPROVED** [review]

**Comment by JordanYates** [comment]
> Would it make sense to add this sort of check to CI?

This PR includes an update to `spelling.txt` to try and catch these in the future.
It's not smart enough to catch a repetition across lines, but its what we have.

**Review by Thalley — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Review by LingaoM — APPROVED** [review]

**Pull request merged**